File: app/src/main/java/fi/kaupunkifillarit/analytics/MapsEvents.java
package fi.kaupunkifillarit.analytics;

import com.google.android.gms.analytics.HitBuilders;

import java.util.Map;

public abstract class MapsEvents {
    private static final String CATEGORY = "Maps";

    public static Map<String, String> myLocationClick() {
        return new HitBuilders.EventBuilder()
                .setCategory(CATEGORY)
                .setAction(Events.ACTION_CLICK)
                .setLabel("My Location")
                .setValue(1)
                .build();
    }
}

File: app/src/main/kotlin/fi/kaupunkifillarit/analytics/MapsEvents.kt
package fi.kaupunkifillarit.analytics

import com.google.android.gms.analytics.HitBuilders

object MapsEvents {
    private const val CATEGORY = "Maps"

    fun myLocationClick(): Map<String, String> {
        return HitBuilders.EventBuilder()
                .setCategory(CATEGORY)
                .setAction(Events.ACTION_CLICK)
                .setLabel("My Location")
                .setValue(1)
                .build()
    }
}
